components/idp-mgt/org.wso2.carbon.idp.mgt/src/test/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAOTest.java

```diff
@@ -0,0 +1,1690 @@
+    public void testGetIdPsSearchWithAttributesException(int tenantId, List<ExpressionNode> expressionNodes, int limit,
+                                                         int offset, String order, String sortBy,
+                                                         List<String> attributes) throws Exception {
+
+        /*
+            Unit Testing for 'getIdPsSearch' method with following parameters
+            getIdPsSearch (int tenantId, List<ExpressionNode> expressionNode, int limit, int offset,
+                           String sortOrder, String sortBy, List<String> requiredAttributes)
+        */
+        mockStatic(IdentityDatabaseUtil.class);
+        IdPManagementDAO idPManagementDAO = new IdPManagementDAO();
+
+        try (Connection connection = getConnection(DB_NAME)) {
+            when(IdentityDatabaseUtil.getDBConnection(anyBoolean())).thenReturn(connection);
+            when(IdentityDatabaseUtil.getDBConnection()).thenReturn(connection);
+            addTestIdps(idPManagementDAO);
+
+            assertThrows(IdentityProviderManagementClientException.class, () -> idPManagementDAO.getIdPsSearch
+                    (tenantId, expressionNodes, limit, offset, order, sortBy, attributes));
+        }
+    }
+
+    @DataProvider
+    public Object[][] getCountOfFilteredIdPsData() {
+
+        ExpressionNode en1 = new ExpressionNode();
+        List<ExpressionNode> expressionNodes1 = new ArrayList<>();
+        expressionNodes1.add(en1);
+        ExpressionNode en2 = new ExpressionNode();
+        en2.setAttributeValue("name");
+        en2.setOperation("sw");
+        en2.setValue("test");
+        List<ExpressionNode> expressionNodes2 = new ArrayList<>();
+        expressionNodes2.add(en2);
+
+        return new Object[][]{
+                {SAMPLE_TENANT_ID, expressionNodes1, 2},
+                {SAMPLE_TENANT_ID, expressionNodes2, 2},
+                {SAMPLE_TENANT_ID2, expressionNodes1, 1},
+        };
+    }
+
+    @Test(dataProvider = "getCountOfFilteredIdPsData")
+    public void testGetCountOfFilteredIdPs(int tenantId, List<ExpressionNode> expNodes, int count) throws Exception {
+
+        mockStatic(IdentityDatabaseUtil.class);
+        IdPManagementDAO idPManagementDAO = new IdPManagementDAO();
+
+        try (Connection connection = getConnection(DB_NAME)) {
+            when(IdentityDatabaseUtil.getDBConnection(anyBoolean())).thenReturn(connection);
+            when(IdentityDatabaseUtil.getDBConnection()).thenReturn(connection);
+            addTestIdps(idPManagementDAO);
+
+            int resultCount = idPManagementDAO.getCountOfFilteredIdPs(tenantId, expNodes);
+            assertEquals(resultCount, count);
+        }
+
+        try {
+            int resultCount = idPManagementDAO.getCountOfFilteredIdPs(tenantId, expNodes);
+        } catch (IdentityProviderManagementServerException e) {
+            assertEquals(e.getErrorCode(), "IDP-65006");
+        }
```

Comment:
What is the purpose of this try block? Can't we merge the catch block with the above try block?

Reply:
Addressed with https://github.com/wso2/carbon-identity-framework/commit/6b030460eec87be9f79276256c9a3617fe357398